app/components/graph-editor/make-draw-node.ts

```diff
@@ -0,0 +1,111 @@
+// File: make-draw-node.ts
+// Created by: CJ Dimaano
+// Date created: January 13, 2016
+//
+// THIS FILE IS INTENDED TO BE IMPORTED ONLY INTO graph-editor.component.ts
+//
+
+
+import * as CONST from "./constants";
+import { GraphEditorCanvas, point } from "./canvas";
+import { DrawableNode } from "./drawable-interfaces";
+
+
+/**
+ * makeDrawNode  
+ *   Makes a draw function for a given node.
+ */
+export function makeDrawNode(canvas: GraphEditorCanvas, g: CanvasRenderingContext2D, n: DrawableNode, dim: any, isDragging: boolean, isHovered: boolean, pt?: point): () => void {
+    // TODO:
+    // Deal with pt.
+    let shadowColor = (isDragging ? CONST.NODE_DRAG_SHADOW_COLOR : (isHovered ? CONST.SELECTION_COLOR : undefined));
```

Comment:
I agree. I'm trying to keep the file short, though, so I don't have to scroll as much.